shared/modules/treestore_module.go

```diff
@@ -31,6 +31,9 @@ type TreeStoreModule interface {
 	Update(pgtx pgx.Tx, height uint64) (string, error)
 	// DebugClearAll completely clears the state of the trees. For debugging purposes only.
 	DebugClearAll() error
+	// Prove generates and verifies a proof against the tree with the matching name using the given
+	// key and value. If value == nil, it will verify non-membership of the key, otherwise membership.
+	Prove(treeName string, key, value []byte) error
```

Comment:
Can you have this return (bool, error) so we can differentiate between errors and just incorrect key-value pairs?